### ISSUE TITLE
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

### ISSUE THREAD

**Issue by Eagle334** [AaronTraas/Clash-Royale-Clan-Tools#14]
Hello, with the latest releases (git pull + sudo python3 setup.py install + crtools), i'm no more able to execute crtools. I have : 
```
$ crtools
custom logo file "/home/eric/myclan/logo.png" not found. Using default instead.
custom favicon file "/home/eric/myclan/favicon.ico" not found. Using default instead.
custom description file "/home/eric/TeamFR/index.html" not found. Using default instead.
DEBUG:crtools.config:{'api': {'server_url': 'https://api.clashroyale.com', 'api_key': 'MyLongAPIKey', 'clan_id': '#Q8JL00'}, 'paths': {'out': '/var/www/html/TeamFR', 'favicon': '/usr/local/lib/python3.6/dist-packages/crtools-1.15.3-py3.6.egg/crtools/templates/crtools-favicon.ico', 'clan_logo': '/usr/local/lib/python3.6/dist-packages/crtools-1.15.3-py3.6.egg/crtools/templates/crtools-logo.png', 'description_html': '~/TeamFR/index.html', 'temp_dir_name': 'crtools', 'description_html_src': None}, 'www': {'canonical_url': 'http://eagluntu.noip.me/TeamFR/'}, 'score': {'min_clan_size': 30, 'war_battle_played': 15, 'war_battle_incomplete': -30, 'war_battle_won': 5, 'war_battle_lost': 0, 'collect_battle_played': 0, 'collect_battle_incomplete': -10, 'collect_battle_won': 3, 'collect_battle_lost': 0, 'war_participation': 0, 'war_non_participation': -5, 'min_donations_daily': 30, 'max_donations_bonus': 50, 'donations_zero': -40, 'threshold_promote': 200, 'threshold_demote': 100, 'threshold_kick': 0, 'threshold_warn': 30}, 'members': {'blacklist': ['#2P8JU82L'], 'vacation': [], 'safe': ['#8GRLU09Q']}, 'crtools': {'debug': True}}
DEBUG:crtools.crtools:crtools version v1.15.3
DEBUG:crtools.crtools:{'api': {'server_url': 'https://api.clashroyale.com', 'api_key': 'MyLongAPIKey', 'clan_id': '#Q8JL00'}, 'paths': {'out': '/var/www/html/TeamFR', 'favicon': '/usr/local/lib/python3.6/dist-packages/crtools-1.15.3-py3.6.egg/crtools/templates/crtools-favicon.ico', 'clan_logo': '/usr/local/lib/python3.6/dist-packages/crtools-1.15.3-py3.6.egg/crtools/templates/crtools-logo.png', 'description_html': '~/TeamFR/index.html', 'temp_dir_name': 'crtools', 'description_html_src': None}, 'www': {'canonical_url': 'http://eagluntu.noip.me/TeamFR/'}, 'score': {'min_clan_size': 30, 'war_battle_played': 15, 'war_battle_incomplete': -30, 'war_battle_won': 5, 'war_battle_lost': 0, 'collect_battle_played': 0, 'collect_battle_incomplete': -10, 'collect_battle_won': 3, 'collect_battle_lost': 0, 'war_participation': 0, 'war_non_participation': -5, 'min_donations_daily': 30, 'max_donations_bonus': 50, 'donations_zero': -40, 'threshold_promote': 200, 'threshold_demote': 100, 'threshold_kick': 0, 'threshold_warn': 30}, 'members': {'blacklist': ['#2P8JU82L'], 'vacation': [], 'safe': ['#8GRLU09Q']}, 'crtools': {'debug': True}}
DEBUG:crtools.api.ClashRoyaleAPI:Retrieving clan data for "#Q8JL00"
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.clashroyale.com
DEBUG:urllib3.connectionpool:https://api.clashroyale.com:443 "GET /v1/clans/%23Q8JL00 HTTP/1.1" 200 None
DEBUG:crtools.api.ClashRoyaleAPI:Retrieving warlog for "#Q8JL00"
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.clashroyale.com
DEBUG:urllib3.connectionpool:https://api.clashroyale.com:443 "GET /v1/clans/%23Q8JL00/warlog HTTP/1.1" 200 None
DEBUG:crtools.api.ClashRoyaleAPI:Retrieving current war data for "#Q8JL00"
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.clashroyale.com
DEBUG:urllib3.connectionpool:https://api.clashroyale.com:443 "GET /v1/clans/%23Q8JL00/currentwar HTTP/1.1" 200 None
Traceback (most recent call last):
  File "/usr/local/bin/crtools", line 11, in <module>
    load_entry_point('crtools==1.15.3', 'console_scripts', 'crtools')()
  File "/usr/local/lib/python3.6/dist-packages/crtools-1.15.3-py3.6.egg/crtools/__init__.py", line 95, in main
    build_dashboard(config)
  File "/usr/local/lib/python3.6/dist-packages/crtools-1.15.3-py3.6.egg/crtools/crtools.py", line 501, in build_dashboard
    old_history = json.loads(myfile.read())
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
... and my index.html is empty.
Any idea ?

**Comment by Eagle334** [comment]
Still the same with this day release :   
```
DEBUG:crtools.api.ClashRoyaleAPI:Retrieving clan data for "#Q8JL00"
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.clashroyale.com
DEBUG:urllib3.connectionpool:https://api.clashroyale.com:443 "GET /v1/clans/%23Q8JL00 HTTP/1.1" 200 None
DEBUG:crtools.api.ClashRoyaleAPI:Retrieving warlog for "#Q8JL00"
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.clashroyale.com
DEBUG:urllib3.connectionpool:https://api.clashroyale.com:443 "GET /v1/clans/%23Q8JL00/warlog HTTP/1.1" 200 None
DEBUG:crtools.api.ClashRoyaleAPI:Retrieving current war data for "#Q8JL00"
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): api.clashroyale.com
DEBUG:urllib3.connectionpool:https://api.clashroyale.com:443 "GET /v1/clans/%23Q8JL00/currentwar HTTP/1.1" 200 None
Traceback (most recent call last):
  File "/usr/local/bin/crtools", line 11, in <module>
    load_entry_point('crtools==1.15.4', 'console_scripts', 'crtools')()
  File "/usr/local/lib/python3.6/dist-packages/crtools-1.15.4-py3.6.egg/crtools/__init__.py", line 95, in main
    build_dashboard(config)
  File "/usr/local/lib/python3.6/dist-packages/crtools-1.15.4-py3.6.egg/crtools/crtools.py", line 519, in build_dashboard
    old_history = json.loads(myfile.read())
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by Eagle334** [comment]
Up, still not working with the commit aa0be20992855001e9f01061ba705daa500d4f1c (HEAD -> master, origin/master, origin/HEAD)
Author: Aaron Traas <aaron@traas.org>
Date:   Tue Mar 19 12:07:53 2019 -0400

**Comment by Eagle334** [comment]
OK, the issue seems to be because i have an empty history.json. 
Now the html is generated, but every players are in red :(

<img width="1237" alt="Capture d’écran 2019-03-22 à 19 40 29" src="https://user-images.githubusercontent.com/37941899/54845733-74f05600-4cda-11e9-857f-0ede22e5ee5b.png">

**Comment by Eagle334** [comment]
Hum, It seems we just have to wait for an update of the players' info. After a new API request I have aone player who has passed "normal". Wait and See...
<img width="1237" alt="Capture d’écran 2019-03-22 à 19 46 18" src="https://user-images.githubusercontent.com/37941899/54846016-314a1c00-4cdb-11e9-8532-9f652fbe51ca.png">

**Comment by Eagle334** [comment]
OK, now everybody is fine. The ticket can be closed, except you wanna manage this case of an empty history.json...

**Comment by AaronTraas** [comment]
This was fixed recently in commit b4267208f7a21b6836ff6986bbc91905a719e014
